go/saltpack/specs/saltpack_armor.md

```diff
@@ -0,0 +1,193 @@
+# SaltPack ASCII Armor Format
+
+Like PGP, SaltPack encrypted and signed messages are binary formats. We need to
+wrap them in an ASCII format, to let people paste them into websites and apps
+that handle text.
+
+The main building block of this format is a generic bytes-to-characters
+encoding, similar to Base64, which we call Just Numbers. We chose not to use
+Base64 mainly because we want to avoid special characters. You should be able
+to paste an encoded message anywhere, without worrying about how punctuation is
+interpreted. Lots of modern applications implement something like Markdown,
+where they interpret slashes and underscores as italics and strip them out.
+Even something as innocuous as `...` might turn into `…` and break decoding.
+
+We define the Just Numbers encoding scheme below, in a way that's independent
+of any particular alphabet or block size. For SaltPack we then select a
+62-character alphabet (all the digits and letters, in ASCII order) and a
+32-byte input block size to use with Just Numbers, which gives us a packing
+efficiency of 74.42%, compared to Base64's 75%. Finally we design some rules
+for spacing and delimiting the output, to make it look nice.
+
+Here's the SaltPack armor encoding of the standard lorem ipsum paragraph:
+
+BEGIN ARMOR.
+I7lAG9yMZ0wAqD2 sOUHnQz6NOuExga WMa8JagoXrIlXO2 cjZukM8JONusQT3 3n7d3nHge2n1aOU AXTTo779PYuO2y7 3We1xXJgVI16Zen TI6UInlynbr9grE zLzZUYe2MQQVEvH g9qDcgNZTKRq61a ynlZYjWZgascz7c 1tDL9iHQBHm3nHl Tsp9RK3xdPabaPU cTKXKxqb1yVd2GB s3dvZNk4QmoAf3L jeZlYvu0GgI7uno EykBRCBDUL8pN2H Lm2P0Lfg8gQdMfs 0jkvLKxuDTaWVvc CZ5l9VGP3pX6VM2 IRMZFIReCt3nv1a oDWMKNeebaEGPpS iXxWsVfZwqUK3K7 h2JMheWsYMsvWPX 8kR2WpulekKqPSH y2tt89L3Qh33RXI vJi305Lc91t7HjJ WQfYWNakquQnjf2 ljGQP9ZqJiRqNA4 3oGpAMOoDuWRSFk 7RdAFeVQDsSr1CD eIFiKcjbNZ6LILU 30a9hAJ9uB0cfVi GpCdR3svMw2Bysi Z5gj8lRqUOGgA57 GWSrfsLlMgZezGI yB0zNqX9SLSpIwI 9l8gOFgoj0BRPGU 0m7OqPgY3zCvlBH 1OmrwpAkA1Wv8.
+END ARMOR.
+
+## Spacing and Delimiting
+
+The decoder has two pre-processing steps:
+
+1. Drop input up to the first period and after the second period. (Note that
+   the second period in the example above comes right *before* "END ARMOR". The
+   final period is ignored.)
+2. Strip all whitespace.
```

Comment:
I don't remember if this has been recorded anywhere, but we want to start as conservative as possible and loosen up as we encounter complaints, rather than going too loose from the beginning. I would say any combo of (whitespace, >) that leads a line should be ignored, for starters.